proto/astarteplatform/msghub/interface.proto

```diff
@@ -23,6 +23,11 @@ syntax = "proto3";
 
 package astarteplatform.msghub;
 
+/* This message defines the introspection a node should send when attaching to the Astarte message hub. */
+message Introspection {
```

Comment:
Yep, I had the same thought, still I decided to differentiate the 2 messages since `Introspection` represents the whole set of interfaces transmitted during the attach procedure, whereas `InterfacesJson` may represent a subset of them or new interfaces to be added to the Node interfaces. Nevertheless, since both the 2 messages represent a set of interfaces, it seems reasonable to use only one of them (the most generic one, thus `InterfacesJson`).

Reply:
@harlem88 @joshuachp what's your opinion?

Reply:
I think that having two structures would be better in this case, so in the future we can add additional fields to the `rcp Attach` message and still be backwards compatible. 

Perhaps the name `Introspection` is confusing in this case, I would still call it Node or a term related to the `Attach` call, with a field called introspection.